src/clj/game/cards/assets.clj

```diff
@@ -1593,28 +1594,33 @@
              :effect (req (gain-tags state :runner eid 1))}]})
 
 (defcard "Malia Z0L0K4"
-  (let [re-enable-target
-        (req (if-let [malia-target (:malia-target card)]
-               (if (:disabled (get-card state malia-target))
-                 (do (system-msg state side (str "uses " (:title card) " to unblank "
-                                                 (card-str state malia-target)))
-                     (enable-card state :runner (get-card state malia-target))
-                     (remove-icon state :runner card (get-card state malia-target))
-                     (if-let [reactivate-effect (:reactivate (card-def malia-target))]
-                       (resolve-ability state :runner eid reactivate-effect (get-card state malia-target) nil)
-                       (effect-completed state nil eid)))
-                 (effect-completed state nil eid))
-               (effect-completed state nil eid)))]
+  (let [unmark
+        (req (when-let [malia-target (get-in card [:special :malia-target])]
+               (update! state side (assoc-in (get-card state card) [:special :malia-target] nil))
+               (remove-icon state :runner card (get-card state malia-target)))
+             ;; I'm not sure why the side is nil here
+             ;; but the old impl had it, so 🤷
+             ;; --nbk, Apr '24
+             (effect-completed state nil eid))]
```

Comment:
for historical reasons, `effect-completed` takes 3 args, but it doesn't use the middle arg, `side`. so as i've been working on the engine, i've been slowly changing that over to be `nil`. i hope to someday completely remove it, but that will also involve removing the `effect` macro, which i don't wanna do yet lol.

Reply:
yeah, I really dislike the effect macro. I feel like it makes the code just that little bit harder to maintain (and it also means you can't throw lets into the code).